sources/web/datalab/polymer/modules/template-manager/template-manager.ts

```diff
@@ -36,44 +34,47 @@ class NotebookTemplate {
   }
 
   /**
-   * Escapes all regex modifier characters in the given string.
+   * Inserts a new cell at position zero that defines python variables
+   * for all of the parameters.
    */
-  private static _regexEscapeAll(s: string) {
-    return s.replace(/([.*+?^${}()|\[\]\/\\])/g, '\\$1');
-  }
-
-  /**
-   * Substitutes all placeholders in the given notebook's cells with their
-   * values.
-   */
-  public populatePlaceholders(notebook: NotebookContent) {
-
-    notebook.cells.forEach((cell: NotebookCell) => {
-      this.parameters.forEach((parameter: TemplateParameter) => {
-        const placeholder = PLACEHOLDER_PREFIX + parameter.placeholder;
-        const escapedPlaceholder = NotebookTemplate._regexEscapeAll(placeholder);
-        const regex = new RegExp(escapedPlaceholder);
-        const value = '\'' + parameter.value.toString() + '\'';
-        cell.source = cell.source.replace(regex, value);
-      });
-    });
-
+  public addParameterCell(notebook: NotebookContent) {
+    const definitionLines = this.parameters.map(parameter =>
+        parameter.placeholder + ' = ' + JSON.stringify(parameter.value));
```

Comment:
Should we rename this to `parameter.name`?

Reply:
Done.